teuthology/provision.py

```diff
@@ -29,105 +30,200 @@ def _get_downburst_exec():
+        self.build_config()
+        success = None
+        with safe_while(sleep=60, tries=3,
+                        action="downburst create") as proceed:
+            while proceed():
+                (returncode, stdout, stderr) = self._run_create()
+                if returncode == 0:
+                    log.info("Downburst created %s: %s" % (self.name,
+                                                           stdout.strip()))
+                    success = True
+                    break
+                elif stderr:
+                    # If the guest already exists first destroy then re-create:
+                    if 'exists' in stderr:
+                        success = False
+                        log.info("Guest files exist. Re-creating guest: %s" %
+                                 (self.name))
+                        self.destroy()
+                    else:
+                        success = False
+                        log.info("Downburst failed on %s: %s" % (
+                            self.name, stderr.strip()))
+                        break
+            return success
+
+    def _run_create(self):
+        """
+        Used by create(), this method is what actually calls downburst when
+        creating a virtual machine instance.
+        """
+        if not self.config_path:
+            raise ValueError("I need a config_path!")
+        shortname = decanonicalize_hostname(self.name)
+
+        args = [self.executable, '-c', self.host, 'create',
+                '--meta-data=%s' % self.config_path, shortname,
+                ]
         log.info("Provisioning a {distro} {distroversion} vps".format(
-            distro=os_type,
-            distroversion=os_version
+            distro=self.os_type,
+            distroversion=self.os_version
         ))
+        proc = subprocess.Popen(args, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+        out, err = proc.communicate()
+        return (proc.returncode, out, err)
+
+    def destroy(self):
+        """
+        Destroy (shutdown and delete) a virtual machine instance.
+        """
+        executable = self.executable
+        if not executable:
+            log.error("No downburst executable found.")
+            return False
+        shortname = decanonicalize_hostname(self.name)
+        args = [executable, '-c', self.host, 'destroy', shortname]
+        proc = subprocess.Popen(args, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,)
+        out, err = proc.communicate()
+        if err:
+            log.error("Error destroying {machine}: {msg}".format(
+                machine=self.name, msg=err))
+            return False
+        elif proc.returncode == 0:
+            out_str = ': %s' % out if out else ''
+            log.info("Destroyed %s%s" % (self.name, out_str))
+            return True
+        else:
+            log.error("I don't know if the destroy of {node} succeded!".format(
```

Comment:
Same as the previous comment (better to avoid first person in logs/messages).

Also, if it is unknown if `destroy` worked, then why is this an error? Maybe it should be a warning?